sdks/go/pkg/beam/core/sdf/lock.go

```diff
@@ -80,3 +80,73 @@ func (rt *LockRTracker) GetRestriction() interface{} {
 	defer rt.Mu.Unlock()
 	return rt.Rt.GetRestriction()
 }
+
+// NewLockBoundableRTracker creates a LockBoundableRTracker initialized with the specified
+// restriction tracker as its underlying restriction tracker.
+func NewLockBoundableRTracker(rt BoundableRTracker) *LockRTracker {
+	return &LockRTracker{Rt: rt}
+}
+
+// LockBoundableRTracker is a restriction tracker that wraps another restriction
+// tracker and adds thread safety to it by locking a mutex in each method,
+// before delegating to the underlying tracker.
+type LockBoundableRTracker struct {
```

Comment:
Is it necessary to have a new type, and duplicating all the methods, instead of adding to the existing RTracker?

Assume as default, if there is no IsBounded method (determinable with a type assertion), that the restriction is bounded, like we would if the restriction/tracker didn't have the method.

Type assertions are fast, and we can always delegate to the underlying tracker when available in the one method.

Reply:
Makes sense. I should have made the changes here as well when doing it for `defaultTruncateRestriction`.